server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverRequest.java

```diff
@@ -67,14 +68,31 @@
             CONDITIONS, ObjectParser.ValueType.OBJECT);
         PARSER.declareField((parser, request, context) -> request.createIndexRequest.settings(parser.map()),
             CreateIndexRequest.SETTINGS, ObjectParser.ValueType.OBJECT);
+        PARSER.declareField((parser, request, includeTypeName) -> {
+            if (includeTypeName) {
+                //expecting one type only
+                for (Map.Entry<String, Object> mappingsEntry : parser.map().entrySet()) {
+                    request.createIndexRequest.mapping((Map<String, Object>) mappingsEntry.getValue());
+                }
+            } else {
+                // a type is not included, add a dummy _doc type
+                Map<String, Object> mappings = parser.map();
+                if (MapperService.isMappingSourceTyped(mappings)) {
```

Comment:
or are you refering to a check on this line ? https://github.com/elastic/elasticsearch/pull/70966/files#r607882926

Reply:
Yes, this check here - I'd just remove it

Reply:
I just realised that removal of this check will make the test in 7.x to fail. Not sure we should remove the check in this case
see the test here:
https://github.com/elastic/elasticsearch/blob/7.x/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.rollover/40_mapping.yml#L46

Reply:
The context: this check was added to fix a bug where you could still pas mappings nested under a type `_doc` when `include_type_name` was `false`. It was specific to the `_doc` type, and arose because of a tricky issue where our request parsing logic doesn't catch it as expected: https://github.com/elastic/elasticsearch/pull/38270.

If we are trying to match the 7.x structure of requests, then I think it makes sense to keep in this fix. (Separately, maybe we should look into making it less intrusive!)

A last note: our original check/ fix was for the `_doc` type only, whereas this attempts a *general* check as to whether the mapping is nested under a type. That is actually not possible in general, as @romseygeek noted in another comment.

Reply:
thanks @jtibshirani 
in this case I will keep the check for the mapping used in compatible api mode but will remove it when a mapping is used in v8